SIH-as-SITL: add  Standard VTOL Airframe

mbjd opened the pull request:
### Solved Problem
We can now simulate all major airframes in [SIH-as-SITL](https://docs.px4.io/main/en/sim_sih/index.html#sih-as-sitl-no-fc) (meaning, both PX4 and PX4-internal sim running on PC). In the future this will allow for depreciation of gazebo in the context of automated tests, making them much faster and simpler. 

### Solution
Add new airframe file, and add it to build system in relevant places. New make target: `make px4_sitl sihsim_svtol`.

### Changelog Entry
For release notes:
```
Add standard VTOL airframe for SIH-as-SITL. 
```

dagar commented:
Not that it's a great name, but I would spell it out as "standard VTOL" everywhere rather than introduce yet another initialism.

Maybe this is just me, but I initially thought it might have been a typo for [STOVL](https://en.wikipedia.org/wiki/STOVL).

mbjd commented:
@dagar good point, agreed. I changed it to standard_vtol in https://github.com/PX4/PX4-Autopilot/pull/24210/commits/a0bf5462bd7b3af2f0e54c653d02921607d7e6e1.

mbjd commented:
Rebased onto latest main which now contains the related https://github.com/PX4/PX4-Autopilot/commit/5bca71791a27db30e89e8904c40cab51e11f8e8b, and squashed my commits into one. Should be ready for merging now.

sfuhrer commented on ROMFS/px4fmu_common/init.d-posix/airframes/10043_sihsim_standard_vtol:
```diff
@@ -0,0 +1,116 @@
+# @output Motor3 MC motor front left
+# @output Motor4 MC motor back right
+# @output Motor5 Forward thrust motor
+# @output Servo1 Aileron
```
```suggestion
# @output Servo1 Ailerons (single channel)
```

sfuhrer commented on ROMFS/px4fmu_common/init.d-posix/airframes/10043_sihsim_standard_vtol:
```diff
@@ -0,0 +1,116 @@
+
+# }}}
+
+param set-default VT_B_TRANS_DUR 5
```
```suggestion
```
We don't need that setting for this airframe, better leave it at default.

sfuhrer commented on ROMFS/px4fmu_common/init.d-posix/airframes/10043_sihsim_standard_vtol:
```diff
@@ -0,0 +1,116 @@
+# }}}
+
+param set-default VT_B_TRANS_DUR 5
+param set-default VT_ELEV_MC_LOCK 0
```
Also here, I would remove it and keep the simulation setup as close to default setup as possible.

sfuhrer commented on ROMFS/px4fmu_common/init.d-posix/airframes/10043_sihsim_standard_vtol:
```diff
@@ -0,0 +1,116 @@
+param set-default CA_SV_CS2_TYPE 4  # rudder
+
+param set-default FW_AIRSPD_MAX 12
+param set-default FW_AIRSPD_MIN 7
```
Does it really stall so low? Could we bump the airspeed range to 10m/s-20m/s, with trim at 15? 
Otherwise we should set EKF2_ARSP_THR to 6, otherwise we don't fuse airspeed.

sfuhrer commented on ROMFS/px4fmu_common/init.d-posix/airframes/10043_sihsim_standard_vtol:
```diff
@@ -0,0 +1,116 @@
+
+# disable some checks to allow to fly:
+# - without real battery
+param set-default CBRK_SUPPLY_CHK 894281
```
So there is no simulated battery?

sfuhrer commented on ROMFS/px4fmu_common/init.d-posix/airframes/10043_sihsim_standard_vtol:
```diff
@@ -0,0 +1,116 @@
+# - without real battery
+param set-default CBRK_SUPPLY_CHK 894281
+# - without safety switch
+param set-default CBRK_IO_SAFETY 22027
```
```suggestion
```
That's already the param default.

sfuhrer reviewed (approved):
I just have some comments about the params. I know that you just copied the settings from the 1103 airframe, so I would actually open a new PR and address it there symmetrically for both 1103 and the new 1043.

mbjd commented:
Closing this in favour of https://github.com/PX4/PX4-Autopilot/pull/24263, which addresses the suggestions by @sfuhrer.
